internal/services/executor/executor.go

```diff
@@ -1159,9 +1163,18 @@ func (e *Executor) taskUpdater(ctx context.Context, et *types.ExecutorTask) {
 	if rt != nil {
 		rt.Lock()
 		// update running task Spec.Stop value only when there's a transitions from false to true,
+		// update running task Spec.Timeout value only when there's a transitions from false to true,
 		// other spec values cannot change once the task has been scheduled
 		if !rt.et.Spec.Stop && et.Spec.Stop {
 			rt.et.Spec.Stop = et.Spec.Stop
+			if rt.et.Spec.StopReason == types.TaskTimedout {
```

Comment:
Just a question: previously we weren't setting `EndTime` here. Why should we set it when stop reason is timedout?